### PR TITLE
Add: server-side grading, fitb working.

### PR DESCRIPTION
**Pull request by bjones1**
A big PR. Most of the code is copied from the existing web2py setup, though.

### CODE REVIEW

**Comment by bjones1** [comment]
I rebased this to fix merge conflicts after my other PRs being accepted.

File: bookserver/routers/auth.py [hunk 1]
```diff
@@ -88,11 +88,12 @@ async def login(
         # alg$salt$hash
         # We need to grab the salt and provide that to the CRYPT function
         # which we import from pydal for now.  Once we are completely off of
-        # web2py then this will change.  The ``web2py_private_key`` is an environment
+        # web2py then this will change. The ``web2py_private_key`` is an environment
```
Comment (bnmnetp) [inline bookserver/routers/auth.py:91]: Dude!  I've been putting 2 spaces after a period since before you were born!

Comment (bjones1) [reply bookserver/routers/auth.py:91]: :) [MLA](https://libguides.css.edu/c.php?g=41682&p=265045) and most modern style guide specify one space. But actually [two spaces is better](https://www.theatlantic.com/science/archive/2018/05/two-spaces-after-a-period/559304/)?

**Comment by bjones1** [comment]
When there are merge conflicts, I've been revbasing to fix them. Do you prefer a merge to fix instead?

**Comment by bjones1** [comment]
I fixed the merge conflicts -- @bnmnetp, this is ready for review.

**Review by bnmnetp — APPROVED** [review]

**Pull request merged**